### PR TITLE
Support Ironpython3.4

### PR DESCRIPTION
**Pull request by chuongmep**
### Purpose

This is an pull request support for [Ironpython 3.4](https://github.com/IronLanguages/ironpython3/releases) release, we had wait along time to can do that.

### Description

Revit_ZnoohL6tsv

### Declarations

Check these if you believe they are true

- [x] This PR fix bug
- [x] This PR for new feature
- [x] The codebase is in a better state after this PR
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] Snapshot of UI changes, if any.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@daren-thomas @eirannejad 

### FYIs

@jeremytammik

### CODE REVIEW

**Comment by chuongmep** [comment]
Please don't merge this at the moment, still testing

**Review by daren-thomas — CHANGES_REQUESTED** [review]
Great work! Exciting!

One minor point: This is a breaking, backwards incompatible change. Python3 is for all intents and purposes a _new_ language and anyone using RPS will have to update all their scripts. We should probably think hard about how to communicate this.

Any ideas? @eirannejad? @jeremytammik?

File: Installer/Installer.cs
Comment (chuongmep) [reply Installer/Installer.cs]: yes, I will change to version 2.0.0

**Comment by jeremytammik** [comment]
> backwards incompatible change, and how to communicate this

Maybe the new version can require a [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix)) header comment. Then it can assume that an old script lacking the comment or specifying a prior version of Python is for Python 2 or previous, try to convert it using [2to3](https://en.wikipedia.org/wiki/History_of_Python#Compatibility) and explain the problem if that fails.

Re. communication: I can post this on The Building Coder blog and in the Revit API discussion group. @eirannejad and others have lots of twitter followers, so we can try to spread the word there and on linkedin as well. Would that suffice?

**Comment by chuongmep** [comment]
I recommend some document user need read after merge this commit to develop and release in master branch : 
- Upgrading from IronPython 2 to 3 : https://github.com/IronLanguages/ironpython3/blob/master/Documentation/upgrading-from-ipy2.md
- differences between IronPython and CPython : https://github.com/IronLanguages/ironpython3/blob/master/Documentation/differences-from-c-python.md
- Main github for IronPython3 : https://github.com/IronLanguages/ironpython3

**Comment by daren-thomas** [comment]
Hmm... the README.md file should probably also be updated (in this PR) to reflect the changes and @chuongmep 's links for further reading also integrated in that updated text.

Since I won't be using this ever, I don't really have a lot of skin in the game. If the three of you are in agreement, I'm fine to merge.

**Comment by chuongmep** [comment]
Hi @daren-thomas , updated to help people can follow with IronPython 3 😉

**Review by daren-thomas — APPROVED** [review]

**Comment by daren-thomas** [comment]
@chuongmep thank you very much!

**Comment by chuongmep** [comment]
@daren-thomas, may be we need merge to master branch from develop to GitHub bot can publish a new version.

**Comment by daren-thomas** [comment]
sure. can you walk me through the process please?

**Comment by chuongmep** [comment]
@daren-thomas  I have create another pull request to do that, just need accept this : 
https://github.com/architecture-building-systems/revitpythonshell/pull/137

**Pull request merged**
